### PR TITLE
CCITCARBON-489 fix ans orch faile. update min python v3.9

### PR DESCRIPTION
**Pull request by guyyaakov1**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -82,6 +82,7 @@ def get_version():
         'Programming Language :: Python :: 3',
         'Programming Language :: Python :: 3 :: Only',
         'Programming Language :: Python :: 3.6',
```
Comment (ryankwilliams) [inline setup.py:84]:
```suggestion
```

This classifier can be removed since the minimum python version supported now is `>=3.9`.

File: setup.py [hunk 2]
```diff
@@ -48,7 +48,7 @@ def get_version():
     packages=find_packages(exclude=['tests*']),
     include_package_data=True,
     zip_safe=False,
-    python_requires=">=3.6",
+    python_requires=">=3.9",
     install_requires=[
         'ansible>=2.5.0',
```
Comment (ryankwilliams) [inline setup.py:53]: While bumping the python version to minimum of 3.9, we probably should update the minimum version for ansible, as 2.5.0 is unmaintained [1]. When installing teflo, it's always going to be pulling in the latest ansible version which as of now is 7.0.0.

```suggestion
        'ansible>=7.0.0',
```

[1] https://docs.ansible.com/ansible/latest/reference_appendices/release_and_maintenance.html#ansible-community-changelogs

File: .github/workflows/run_unit_tests.yml [hunk 3]
```diff
@@ -18,24 +18,24 @@ jobs:
     strategy:
       matrix:
         container:
-        - 'centos:7'
+        - 'quay.io/gyaakov/ubi_py39'
```
Comment (ryankwilliams) [inline .github/workflows/run_unit_tests.yml:21]: @guyyaakov1 I see you are using a custom container image maintained by you. Just wondering why is that? Does the _ubi8/python-39_ not provide packages this job requires?

https://catalog.redhat.com/software/containers/ubi8/python-39/6065b24eb92fbda3a4c65d8f

Comment (guyyaakov1) [reply .github/workflows/run_unit_tests.yml]: Thanks For the help!
Will let you know when i will update this images.

Comment (rnetser) [reply .github/workflows/run_unit_tests.yml]: @guyyaakov1 @ryankwilliams any reason why not to use centos-stream9?

Comment (guyyaakov1) [reply .github/workflows/run_unit_tests.yml]: @rnetser is this img need to be pulled with docker creds?

Comment (rnetser) [reply .github/workflows/run_unit_tests.yml]: This is the image `quay.io/centos/centos:stream9`

Comment (ryankwilliams) [reply .github/workflows/run_unit_tests.yml]: @rnetser No specific reason, I saw @guyyaakov1 had `ubi` within his container image name and just was providing the standard ubi8 or even ubi9 image that could be used.

File: tests/functional/test_core_classes.py [hunk 4]
```diff
@@ -172,10 +172,10 @@ def cleanup_master():
 
 @pytest.fixture
 def create_group_vars():
-    gv_path = os.path.join('/tmp/ws/ansible/group_vars', 'ans_gv')
+    gv_path = os.path.join('/tmp/ws/ansible/group_vars/ans_gv.yml')
```
Comment (rujutashinde) [inline tests/functional/test_core_classes.py:175]: this will not work.
you are not joining anything here

**Review by rujutashinde — CHANGES_REQUESTED** [review]

File: tests/functional/test_teflo.py [hunk 5]
```diff
@@ -46,6 +46,13 @@ def runner():
     return CliRunner()
 
 
+@pytest.fixture(scope='function')
+def cleanup_master(request):
```
Comment (rujutashinde) [inline tests/functional/test_teflo.py:50]: @guyyaakov1 i am not sure how you are using this fixture, it is calling cleanup_master() within he cleanup_master(request) method?

File: teflo/orchestrators/ext/ansible_orchestrator_plugin/ansible_orchestrator_plugin.py [hunk 6]
```diff
@@ -129,7 +129,7 @@ def validate(self):
                                                  self.playbook.get('name'))
 
     def __playbook__(self):
-        self.logger.info('Executing playbook:')
+        # self.logger.info('Executing playbook:')
```
Comment (rujutashinde) [inline teflo/orchestrators/ext/ansible_orchestrator_plugin/ansible_orchestrator_plugin.py:132]: remove this comment

**Review by rujutashinde — CHANGES_REQUESTED** [review]

File: .github/workflows/run_unit_tests.yml [hunk 7]
```diff
@@ -18,33 +18,41 @@ jobs:
...
       # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - run: cat /etc/os-release
 
       # Runs a single command using the runners shell
       - name: install required packages
         run: |
-          yum install -y gcc python3 python3-devel python3-pip krb5-workstation krb5-devel git
+          yum install -y glibc-langpack-en gcc python3 python3-devel python3-pip krb5-workstation krb5-devel git
```
Comment (ryankwilliams) [inline .github/workflows/run_unit_tests.yml:35]:
```suggestion
          dnf install -y glibc-langpack-en gcc python3 python3-devel python3-pip krb5-workstation krb5-devel git
```

**Review by rujutashinde — APPROVED** [review]
LGTM

**Pull request merged**